### DISCUSSION TITLE
Looking to get involved

### DISCUSSION THREAD

**Discussion by georgemarklow** [mathnet/mathnet-numerics#1101] [General]
Hi Team

I'm looking for a GitHub project related to scientific computing to get involved in and considering a few options. I'm a C#/TypeScript developer with a bachelor's degree in maths, and about 12 years of coding experience, interested in linear algebra, interpolation, root-finding, calculus and statistics topics. Are there any additional features or upgrades that are needed in mathnet-numerics C# projects? I did have a question if anybody was looking for a TypeScipt port of the code with unit testing, as I really enjoy working with TypeScript too.

I will apply for part-time/distance master's degree programmes in scientific computing as a mature student over the next year or so (I graduated over a decade ago). Any contributions I can make to the open-source community will help strengthen my application as a mature student.

I look forward to hearing from you.

Thanks vm
George

**Comment by strMikhailPotapenko** [comment]
Hello George. I have similar interests to you and made a pull request related to a work project last year. It seems that the maintainer of this repo has not been active since 2023 so that PR was not merged. Obviously, they are under no obligation to continue to maintain the library so it's totally understandable. However, I was disappointed because I use this library for my work and would like to see it continue to grow and have a community around it.

Pursuant of that my company, Corning Inc, is going to be forking this repo and maintaining it from within our Github organization. We are in the process of working this out internally before we can make our fork public on Github. If you are still interested, we would love to work with you and I believe there are a number of features that have already been discussed here or new ones that we can brainstorm that would be good additions to the library.

If you like, I can reach out to you again once our fork is public to talk more!

-Mikhail

**Comment by jkalias** [comment]
Hi Mikhail,

Please let me know if/when your company decides to fork this repo, I would love to contribute there as well, since as you have said this repo seems to be stale for quite some time.

-John

**Comment by strMikhailPotapenko** [comment]
Hello John,

We would love to have your contributions and I will absolutely let you know once we are ready! If I'm not mistaken, you are the current maintainer of MathNet Spatial? I have to admit that I haven't maintained an open source project like this before although I do have some experience in leading and working on closed source development teams. I would welcome any suggestions and feedback.

One question for you on that note: are you opposed in principle to signing a CLA? 

My company's current policy requires this and we have been having some discussions internally about it. I can imagine that due to the nature of open source contribution culture, some people may be hesitant to sign a CLA with a corporation. Do you find in your experience that this could be a barrier for getting community engagement?

-Mikhail

**Comment by jkalias** [comment]
Hi Mikhail,

Yes, I am one of the maintainers of MathNET.Spatial, even though I don't have rights to publish updated Nuget packages, or even change/update the direction of the project as I experienced here https://github.com/mathnet/mathnet-spatial/pull/229.

I have no experience with CLA, neither on a personal level nor about what other peoples' feelings are about such an agreement. Honestly I would be a bit hesitant to contribute to any project which requires me to sign any agreement in that case.

Hope this helps, best regards
-John

**Comment by strMikhailPotapenko** [comment]
Hello John,

First of all, love seeing the reference to Geometric Algebra in that discussion. I did have some thoughts previously as to whether it would be worth putting any GA methods into the spatial library at some point as my understanding for some tasks a GA based API can be cleaner (but of course that's a discussion for another time and place).

Second, thank you for your feedback on the CLA, I will share that internally. One follow up question if you don't mind: is there anything in the way of explanation of the CLA and/or the intent behind it that we could do to ease your apprehensions about contributing? I understand why you might feel hesitant and to be honest am worried many other developers would feel the same way so this seems like an important question to me.

-Mikhail

**Reply by jkalias** [reply]
> Hello John,
> 
> First of all, love seeing the reference to Geometric Algebra in that discussion. I did have some thoughts previously as to whether it would be worth putting any GA methods into the spatial library at some point as my understanding for some tasks a GA based API can be cleaner (but of course that's a discussion for another time and place).

I have to admit that I am late to the game of Geometric Algebra, I discovered it more or less a couple of years ago. However, I think its potential is HUG for stable numerical computing. But as you rightly point out, this is another topic of its own.

> Second, thank you for your feedback on the CLA, I will share that internally. One follow up question if you don't mind: is there anything in the way of explanation of the CLA and/or the intent behind it that we could do to ease your apprehensions about contributing? I understand why you might feel hesitant and to be honest am worried many other developers would feel the same way so this seems like an important question to me.
> 
> -Mikhail

My main line of thinking is, that people who are contributing to open source are doing so, because they feel pationate about a certain topic and want to share their knowledge and/or experience with other people. I understand that there are also companies heavily involved in the OSS community as well (for example Apple with its Swift approach or Microsoft with .NET), however I am not really comfortable signing a document from ANY company regarding open source, since I am not the employee of this company. If they decide to sponsor my contributions, that's a whole other topic, but basically it's my free time which I am sacrificing.

Maybe this helps a bit more.

**Reply by Arlofin** [reply]
Hello all,

is there a chance to get a new maintainer? This would be my preference over the creation of forks. After all, this library is well-known and established, but it needs further evolution.

**Reply by strMikhailPotapenko** [reply]
That would be a good option but the previous maintainer would have to grant rights. I reached out on Mastodon to ask if he would be willing to pass the rights over but received no reply. Mastodon is the only way I could find to contact him but would be happy to try other suggestions.
